third_party/terraform/utils/config_test.go

```diff
@@ -90,6 +90,34 @@ func TestAccConfigLoadValidate_credentials(t *testing.T) {
 	}
 }
 
+func TestAccConfigLoadValidate_impersonated(t *testing.T) {
+	if os.Getenv(TestEnvVar) == "" {
+		t.Skip(fmt.Sprintf("Network access not allowed; use %s=1 to enable", TestEnvVar))
+	}
+	testAccPreCheck(t)
+
+	serviceaccount := multiEnvSearch([]string{"GOOGLE_IMPERSONATE_SERVICE_ACCOUNT"})
+	proj := getTestProjectFromEnv()
+
+	config := &Config{
+		ImpersonateServiceAccount: serviceaccount,
+		Project:                   proj,
+		Region:                    "us-central1",
+	}
+
+	ConfigureBasePaths(config)
+
+	err := config.LoadAndValidate(context.Background())
+	if err != nil {
+		t.Fatalf("error: %v", err)
+	}
+
+	_, err = config.clientCompute.Zones.Get(proj, "us-central1-a").Do()
```

Comment:
Hi @upodroid - looks like we have the same issue here as we have in the other PR, this should be updated to use the `NewComputeClient(config.userAgent)` method.

Reply:
fixed